library/ssl_tls13_generic.c

```diff
@@ -2944,14 +2946,14 @@ static int ssl_new_session_ticket_parse( mbedtls_ssl_context* ssl,
 
         memcpy( ssl->session->ticket_nonce, &buf[9], ssl->session->ticket_nonce_len );
 
-        MBEDTLS_SSL_DEBUG_BUF( 3, "ticket->nonce:", (unsigned char*)&buf[9],
-        ssl->session->ticket_nonce_len );
+        MBEDTLS_SSL_DEBUG_BUF( 3, "ticket->nonce:", &buf[9],
+                               ssl->session->ticket_nonce_len );
 
     }
 
     /* Ticket */
-    ticket_len = ( buf[9 + ssl->session->ticket_nonce_len] << 8 ) |
-                 ( buf[10 + ssl->session->ticket_nonce_len] );
+    ticket_len = ( (size_t) buf[9 + ssl->session->ticket_nonce_len] << 8 ) |
```

Comment:
I think the code would be more readable without those magic offset constants. We can either increment the buffer pointer as we process the ticket (preferred on first sight), or have constants for the offsets which are defined next to the documentation of how the ticket is structured.